src/low-level/imap/mailimap_parser.c

```diff
@@ -7885,9 +7917,17 @@ mailimap_msg_att_rfc822_parse_progress(mailstream * fd, MMAPString * buffer, str
   r = mailimap_space_parse(fd, buffer, &cur_token);
   if (r != MAILIMAP_NO_ERROR)
     return r;
-  
+
+  if (parser_ctx) {
```

Comment:
Do you really need to check parser_ctx? It sounds like we can make assumption it's always != NULL.